### PR TITLE
performance optimize meson

### PR DESCRIPTION
**Pull request by marcelhollerbach**
Before this commit meson took on my laptop ~30-40 sec. now we are down to ~20-25 sec..
Those numbers are from building EFL with meson. We are working a lot with generated sources, which are added as source to dependencies, hence we have a big number of include directories, and custom targets.

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -2011,6 +2011,27 @@ def get_custom_target_dir_include_args(self, target, compiler):
...
+        # created for some source dir. For example if someone does this:
+        #
+        # inc = include_directories('foo/bar/baz')
+        #
+        # But never subdir()s into the actual dir.
+        if os.path.isdir(os.path.join(self.environment.get_build_dir(), expdir)):
+            bargs = compiler.get_include_args(expdir, is_system)
+        else:
+            bargs = []
+        return (sargs,bargs)
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[Flake8]__ [E231] missing whitespace after ',' ([link](https://sider.review/gh/repos/19784232/pulls/5440#issue-4157991))

<sub>posted by Sider</sub>

**Comment by marcelhollerbach** [comment]
I really don't understand why those windows builds are missing, as far as i can see, my patch does not change any output, so why is this failing ? Can someone help me here?

**Comment by mensinda** [comment]
The CI is most likely broken. See #5436.

**Comment by marcelhollerbach** [comment]
Oh, i did not see that, thank you @mensinda.

**Pull request merged**